ttnn/cpp/ttnn/operations/data_movement/reshape_on_device/device/reshape_op.cpp

```diff
@@ -26,10 +26,10 @@ void ReshapeDeviceOperation::validate(const std::vector<Tensor>& input_tensors)
 
     TT_FATAL(
         input_tensor_a.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED,
-        "Reshape does not currently support sharding");
+        "Use ttnn::experimental::unsafe_view for reshaping sharded inputs");
     TT_FATAL(
         this->output_mem_config.memory_layout == TensorMemoryLayout::INTERLEAVED,
-        "Reshape does not currently support sharding");
+        "Use ttnn::experimental::unsafe_view for reshaping sharded inputs");
```

Comment:
I think old error message is better. It tells you users what's wrong with current path. Advocating for another op inside an op seems weird as an error message.